src/Extension/FluentExtension.php

```diff
@@ -835,4 +846,106 @@ public function cacheKeyComponent()
     {
         return 'fluentlocale-' . FluentState::singleton()->getLocale();
     }
+
+    /**
+     * @param FieldList $fields
+     */
+    public function updateCMSFields(FieldList $fields)
+    {
+        // get all fields to translate and remove
+        $translated = $this->getLocalisedTables();
+
+        $currentLocale = Locale::getCurrentLocale();
+        $locale = $currentLocale->Locale;
+        $language = $currentLocale ? strtok($locale, '_') : '';
+
+        foreach ($translated as $table => $translatedFields) {
+            foreach ($translatedFields as $translatedField) {
+                // Find field matching this translated field
+                // If the translated field has an ID suffix also check for the non-suffixed version
+                // E.g. UploadField()
+                $field = $fields->dataFieldByName($translatedField);
+                if (!$field && preg_match('/^(?<field>\w+)ID$/', $translatedField, $matches)) {
+                    $field = $fields->dataFieldByName($matches['field']);
+                }
+
+                // Highlight any translatable field
+                if ($field && !$field->hasClass('LocalisedField')) {
+                    // Add a language indicator next to the fluent icon
+                    $title = $field->Title();
+
+                    $titleClasses = 'fluent-locale-label';
+                    $modifiedTitle = 'Using default locale value';
+
+                    $field->setTitle(
+                        DBHTMLText::create()->setValue(
+                            sprintf(
+                                '<span class="%s" title="%s">%s</span>%s',
+                                $titleClasses,
+                                $modifiedTitle,
+                                $language,
+                                $title
+                            )
+                        )
+                    );
+
+                    $field->addExtraClass('LocalisedField');
+                }
+            }
+        }
+
+        $this->addLocaleIndicatorMessage($fields);
+    }
+
+    /**
+     * Adds a UI message to indicate whether you're editing in the default locale or not
+     *
+     * @param  FieldList $fields
+     * @return $this
+     */
+    protected function addLocaleIndicatorMessage(FieldList $fields)
+    {
+        if (Config::inst()->get(__CLASS__, 'disable_current_locale_message')) {
+            return $this;
+        }
+
+        // If the field is already present, don't add it a second time
+        if ($fields->fieldByName('CurrentLocaleMessage')) {
+            return $this;
+        }
+
+        $localeNames     = Locale::getCached()->map('Locale', 'Title')->toArray();
+        $defaultLocale   = Locale::getDefault();
+        $isDefaultLocale = ($defaultLocale === Locale::getCurrentLocale());
```

Comment:
You can use `$isDefaultLocale = $currentLocale->getIsDefault()` rather than performing this logic again